src/auth/crypto.cpp

```diff
@@ -258,8 +258,34 @@ PasswordHashAlgorithm &InternalCurrentHashAlgorithm() {
   std::call_once(flag, [] { current = InternalParseHashAlgorithm(FLAGS_password_encryption_algorithm); });
   return current;
 }
+
+std::optional<std::string_view> UsesAlgo(std::string_view str, PasswordHashAlgorithm algo) {
+  // header = algo name + :
+  const auto header = std::string{AsString(algo)} + ":";
```

Comment:
The `AsString` responses need to be made absolutely stable, and publicly documented